spec/integration/use_cases/accounts/apps/evolution_api/webhooks/evolution_apis_controller_spec.rb

```diff
@@ -0,0 +1,102 @@
+  let(:account) { create(:account) }
+  let(:evolution_api) { create(:apps_evolution_api, account: account) }
+  let(:evolution_api_connected) { create(:apps_evolution_api, :connected, account: account) }
+  let(:evolution_api_connecting) { create(:apps_evolution_api, :connecting, account: account) }
+  let(:qrcode_updated_webhook_event) { load_webhook_event('qrcode_updated_event.json') }
+  let(:created_connection_event) { load_webhook_event('created_connection_event.json') }
+  let(:deleted_connection_event) { load_webhook_event('deleted_connection_event.json') }
+  let(:delete_instance_response) { File.read("spec/integration/use_cases/accounts/apps/evolution_api/instance/delete_response.json") }
+  let(:evolution_api_qrcode_info) { evolution_api.reload.qrcode_info }
+
+
+  def load_webhook_event(filename)
+    File.read(Rails.root.join('spec/integration/use_cases/accounts/apps/evolution_api/webhooks/events', filename))
+  end
+
+  def post_webhook(event_data)
+    Sidekiq::Testing.inline! { post '/apps/evolution_apis/webhooks', params: JSON.parse(event_data) }
+  end
+
+  def expect_success
+    expect(response).to have_http_status(200)
+    expect(JSON.parse(response.body)).to eq({ 'ok' => true })
+  end
+
+  describe 'POST /apps/evolution_apis/webhooks' do
+    describe 'when is qrcode_update event' do
+      context 'when evolution_api is connecting' do
+        it 'should update qrcode' do
+          evolution_api_connecting
+          post_webhook(qrcode_updated_webhook_event)
+          expect_success
+          expect(evolution_api_connecting.reload.qrcode).to eq('qrcode')
+        end
+      end
+      context 'when evolution_api is disconnected' do
+        it 'evolution_api should not be changed' do
+          evolution_api
+          post_webhook(qrcode_updated_webhook_event)
+          expect_success
+          expect(evolution_api.reload.changed?).to be_falsey
+        end
+      end
+      context 'when evolution_api is connected' do
+        it 'evolution_api should not be changed' do
+          evolution_api_connected
+          post_webhook(qrcode_updated_webhook_event)
+          expect_success
+          expect(evolution_api_connected.reload.changed?).to be_falsey
+        end
+      end
+    end
+    describe 'when is created_connection event' do
+      context 'success' do
+        it 'should update evolution_api status, phone and qrcode' do
+          evolution_api.update(qrcode: 'qrcode')
+          post_webhook(created_connection_event)
+          expect_success
+          expect(evolution_api.reload.connected?).to be_truthy
+          expect(evolution_api.phone).to be_present
+          expect(evolution_api.qrcode).not_to be_present
+        end
+      end
+    end
+    describe 'when is deleted_connection event' do
+      before do
+        stub_request(:delete, /delete/)
+        .to_return(body: delete_instance_response.to_json, status: 200, headers: {'Content-Type' => 'application/json'})
+      end
```

Comment:
Esse stub pelo que entendi deve ser rodado somente no `when evolution_api is connected` e `when evolution_api is connecting` não no `when evolution_api is disconnected`
Ai sugiro ativar ele somente no contexto que precisa para facilitar o entendimento do teste também validar melhor o comportamento

Reply:
Show! Separei nos testes de connected e connecting.